pkg/controller/kubelet-config/kubelet_config_bootstrap.go

```diff
@@ -0,0 +1,165 @@
+package kubeletconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	kubeletCfgMasterLable = "pools.operator.machineconfiguration.openshift.io/master"
+	kubeletCfgWorkerLable = "pools.operator.machineconfiguration.openshift.io/worker"
+)
+
+// RunKubeletBootstrap generates MachineConfig objects for mcpPools that would have been generated by syncKubeletConfig
+func RunKubeletBootstrap(templateDir string, kubeletConfigs []*mcfgv1.KubeletConfig, controllerConfig *mcfgv1.ControllerConfig, features *configv1.FeatureGate, mcpPools []*mcfgv1.MachineConfigPool) ([]*mcfgv1.MachineConfig, error) {
+	generatedManagedKeys := make(map[string]string)
+	generatedMCs := make(map[string]*mcfgv1.MachineConfig)
+	// Validate the KubeletConfig CR if exists
+	for _, kubeletConfig := range kubeletConfigs {
+		if kubeletConfig == nil {
+			continue
+		}
+		if err := validateUserKubeletConfig(kubeletConfig); err != nil {
+			return nil, newForgetError(err)
+		}
+	}
+
+	userDefinedSystemReserved := make(map[string]string)
+	var res []*mcfgv1.MachineConfig
+	for _, kubeletConfig := range kubeletConfigs {
+		if kubeletConfig == nil || kubeletConfig.Spec == (mcfgv1.KubeletConfigSpec{}) {
+			continue
+		}
+		for _, pool := range mcpPools {
+			// skip the pool if no default label for kubeletconfig
+			_, hasMasterLabel := pool.Labels[kubeletCfgMasterLable]
+			_, hasWorkerLabel := pool.Labels[kubeletCfgWorkerLable]
+			if !hasMasterLabel && !hasWorkerLabel {
+				continue
+			}
+			role := pool.Name
+
+			originalKubeConfig, err := generateOriginalKubeletConfigWithFeatureGates(controllerConfig, templateDir, role, features)
+			if err != nil {
+				return nil, err
+			}
+			if kubeletConfig.Spec.TLSSecurityProfile != nil {
+				// Inject TLS Options from Spec
+				observedMinTLSVersion, observedCipherSuites := getSecurityProfileCiphers(kubeletConfig.Spec.TLSSecurityProfile)
+				originalKubeConfig.TLSMinVersion = observedMinTLSVersion
+				originalKubeConfig.TLSCipherSuites = observedCipherSuites
+			}
```

Comment:
This being outside of `generateKubeletIgnFiles` implies that this could be different between day 1 and day 2, is that correct? Any reason not to include in `generateKubeletIgnFiles` and remove from here and the day 2 controller?

Reply:
The logic to get the security profile that should be passed into `getSecurityProfileCiphers` is different I have't fonud a good way to merge the code into `generateKubeletIgnFiles` https://github.com/openshift/machine-config-operator/blob/4ad850b8649af61992796dcf923d986861bd82ab/pkg/controller/kubelet-config/kubelet_config_controller.go#L549